UIKit/Location Controller/TWTLocationController.m

```diff
@@ -0,0 +1,189 @@
+//
+//  TWTLocationController.m
+//  Toast
+//
+//  Created by Steve Foster on 2/3/15.
+//  Copyright (c) 2015 Two Toasters, LLC. All rights reserved.
+//
+
+#import "TWTLocationController.h"
+
+@import CoreLocation;
+
+static NSTimeInterval const kLocationTimeoutInterval = 60;
+NSString *const kLocationControllerErrorDomain = @"locationcontroller";
+
+
+@interface TWTLocationController () <CLLocationManagerDelegate>
+
+@property (nonatomic, strong) CLLocationManager *locationManager;
+@property (nonatomic, strong) CLLocation *location;
+@property (nonatomic, strong) NSMutableArray *requestBlocks;
+
+@end
+
+@implementation TWTLocationController
+
++ (instancetype)sharedInstance
+{
+    static dispatch_once_t onceToken;
+    static TWTLocationController *sharedInstance = nil;
+    dispatch_once(&onceToken, ^{
+        sharedInstance = [[TWTLocationController alloc] init];
+    });
+    return sharedInstance;
+}
+
+
+- (instancetype)init
+{
+    self = [super init];
+    if (self) {
+        self.locationManager = [[CLLocationManager alloc] init];
+        self.locationManager.delegate = self;
+        self.locationManager.desiredAccuracy = kCLLocationAccuracyHundredMeters;
```

Comment:
While that’s definitely the easiest solution, you need to be sure external users don’t become the delegate, as that will break this entire class.

You might be better served by having properties on your location controller which basically just pass through the settings to your CLLocationManager. I realize this is more code, but it seems like what you actually want.